test/integration-tests/Uniswap.test.ts

```diff
@@ -1196,6 +1206,480 @@ describe('Uniswap V2 and V3 Tests:', () => {
     })
   })
 
+  describe('Migrator', () => {
+    beforeEach(async () => {
+      // Bob max-approves the v3PM to access his USDC and WETH
+      await usdcContract.connect(bob).approve(v3NFTPositionManager.address, MAX_UINT)
+      await wethContract.connect(bob).approve(v3NFTPositionManager.address, MAX_UINT)
+
+      let bobUSDCBalanceBefore = await usdcContract.balanceOf(bob.address)
+      let bobWETHBalanceBefore = await wethContract.balanceOf(bob.address)
+
+      // need to mint the nft to bob
+      const tx = await v3NFTPositionManager.mint({
+        token0: USDC.address,
+        token1: WETH.address,
+        fee: FeeAmount.LOW,
+        tickLower: 0,
+        tickUpper: 194980,
+        amount0Desired: expandTo6DecimalsBN(2500),
+        amount1Desired: expandTo18DecimalsBN(1),
+        amount0Min: 0,
+        amount1Min: 0,
+        recipient: bob.address,
+        deadline: MAX_UINT,
+      })
+
+      let bobUSDCBalanceAfter = await usdcContract.balanceOf(bob.address)
+      let bobWETHBalanceAfter = await wethContract.balanceOf(bob.address)
+
+      let usdcSpent = bobUSDCBalanceBefore.sub(bobUSDCBalanceAfter)
+      let wethSpent = bobWETHBalanceBefore.sub(bobWETHBalanceAfter)
+
+      // check that the USDC and WETH were spent
+      expect(usdcSpent).to.be.gte(0)
+      expect(wethSpent).to.be.gte(0)
+
+      const receipt = await tx.wait()
+
+      const transferEvent = receipt.events?.find((event) => event.event === 'IncreaseLiquidity')
+
+      tokenId = transferEvent?.args?.tokenId
+    })
+
+    describe('erc721permit', () => {
+      it('erc721 permit succeeds', async () => {
+        const { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+
+        planner.addCommand(CommandType.ERC721_PERMIT, [ADDRESS_THIS, tokenId, MAX_UINT, v, r, s])
+
+        // bob permits the router to spend token
+        await executeRouter(planner)
+
+        expect((await v3NFTPositionManager.positions(tokenId)).operator).to.eq(router.address)
+      })
+
+      it('erc721 permit also succeeds when passing in actual address', async () => {
+        const { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+
+        planner.addCommand(CommandType.ERC721_PERMIT, [router.address, tokenId, MAX_UINT, v, r, s])
+
+        // bob permits the router to spend token
+        await executeRouter(planner)
+
+        expect((await v3NFTPositionManager.positions(tokenId)).operator).to.eq(router.address)
+      })
+
+      it('only owner of the token can permit another address', async () => {
```

Comment:
The reason this test fails is because eve isnt the owner of bob's token. So the signature over `tokenId` which is bob's doesnt work because she doesnt own it.
However I think if you add a test where Bob generates a signature, and eve submits bob's signature it will succeed in approving UR for bob's token. This is a correct behaviour but would be good to add a test to show it.

I think good to have both test cases - where 1 fails and the other passes and make it clear the difference between the 2 tests